src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchResultRendererTable.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using NuGet.Configuration;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.CommandLine.XPlat
+{
+    internal class PackageSearchResultRendererTable : IPackageSearchResultRenderer
+    {
+        private PackageSearchArgs _args;
+        private const int LineSeparatorLength = 40;
+        private static readonly string SourceSeparator = new('*', LineSeparatorLength);
+
+        public PackageSearchResultRendererTable(PackageSearchArgs packageSearchArgs)
```

Comment:
You just need the logger and the search term correct?
Why not pass those 2? 
No need to introduce a dependency between the search args and the renderer.